### ISSUE TITLE
The error in FP8-quantization/quantization/quantizers /fp8_quantizer.py

### ISSUE THREAD

**Issue by Barnonewdm** [Qualcomm-AI-research/FP8-quantization#2]
https://github.com/Qualcomm-AI-research/FP8-quantization/blob/ede19a9bbe603edf9ec106af7545d9691b3a350c/quantization/quantizers/fp8_quantizer.py#L128C5-L128C92

This line should be 
`log_scales = torch.clamp((torch.floor(torch.log2(torch.abs(xc))) + bias).detach(), 1.0)`

**Comment by TiRune** [comment]
Could you explain a bit further why you think this line should be different?

**Comment by Barnonewdm** [comment]
`log_scales = torch.clamp((torch.floor(torch.log2(torch.abs(xc)) + bias)).detach(), 1.0)`

`scales = 2.0 ** (log_scales - M - bias)`

`result = round_ste_func(xc / scales) * scales`
 
Your design is good after reviewing the code. Bias could be a fractional number in your design. 

Initially, my understanding is that the `result` should be on the FP8 grid point with integer bias.
